pace-util/pace/util/grid/helper.py

```diff
@@ -0,0 +1,513 @@
+        horizontal_data: HorizontalGridData,
+        vertical_data: VerticalGridData,
+        contravariant_data: ContravariantGridData,
+        angle_data: AngleGridData,
+    ):
+        self._horizontal_data = horizontal_data
+        self._vertical_data = vertical_data
+        self._contravariant_data = contravariant_data
+        self._angle_data = angle_data
+
+    @classmethod
+    def new_from_metric_terms(cls, metric_terms: MetricTerms):
+        # TODO fix <Quantity>.storage mask for FieldI
+        shape = metric_terms.lon.data.shape
+        edge_n = utils.make_storage_data(
+            metric_terms.edge_n.data,
+            (shape[0],),
+            axis=0,
+            backend=metric_terms.edge_n.gt4py_backend,
+        )
+        edge_s = utils.make_storage_data(
+            metric_terms.edge_s.data,
+            (shape[0],),
+            axis=0,
+            backend=metric_terms.edge_s.gt4py_backend,
+        )
+        edge_e = utils.make_storage_data(
+            metric_terms.edge_e.data,
+            (1, shape[1]),
+            axis=1,
+            backend=metric_terms.edge_e.gt4py_backend,
+        )
+        edge_w = utils.make_storage_data(
+            metric_terms.edge_w.data,
+            (1, shape[1]),
+            axis=1,
+            backend=metric_terms.edge_w.gt4py_backend,
+        )
+
+        horizontal_data = HorizontalGridData(
+            lon=metric_terms.lon.storage,
+            lat=metric_terms.lat.storage,
+            lon_agrid=metric_terms.lon_agrid.storage,
+            lat_agrid=metric_terms.lat_agrid.storage,
+            area=metric_terms.area.storage,
+            area_64=metric_terms.area.storage,
+            rarea=metric_terms.rarea.storage,
+            rarea_c=metric_terms.rarea_c.storage,
+            dx=metric_terms.dx.storage,
+            dy=metric_terms.dy.storage,
+            dxc=metric_terms.dxc.storage,
+            dyc=metric_terms.dyc.storage,
+            dxa=metric_terms.dxa.storage,
+            dya=metric_terms.dya.storage,
+            rdx=metric_terms.rdx.storage,
+            rdy=metric_terms.rdy.storage,
+            rdxc=metric_terms.rdxc.storage,
+            rdyc=metric_terms.rdyc.storage,
+            rdxa=metric_terms.rdxa.storage,
+            rdya=metric_terms.rdya.storage,
+            a11=metric_terms.a11.storage,
+            a12=metric_terms.a12.storage,
+            a21=metric_terms.a21.storage,
+            a22=metric_terms.a22.storage,
+            edge_w=edge_w,
+            edge_e=edge_e,
+            edge_s=edge_s,
+            edge_n=edge_n,
+        )
+        ak = metric_terms.ak.data
+        bk = metric_terms.bk.data
+        # TODO fix <Quantity>.storage mask for FieldK
+        ak = utils.make_storage_data(
+            ak, ak.shape, len(ak.shape) * (0,), backend=metric_terms.ak.gt4py_backend
+        )
+        bk = utils.make_storage_data(
+            bk, bk.shape, len(bk.shape) * (0,), backend=metric_terms.ak.gt4py_backend
+        )
```

Comment:
Similarly, can we do this without `make_storage_data`?

Reply:
Fixed. Added an optional mask input to Quantity to have mask (True, True, False), then `.storage` works as expected.